kml/serdes_gpx.cpp

```diff
@@ -140,6 +141,35 @@ void GpxParser::ParseColor(std::string const & value)
   m_color = kml::ToRGBA(colorBytes[0], colorBytes[1], colorBytes[2], (char)255);
 }
 
+// https://osmand.net/docs/technical/osmand-file-formats/osmand-gpx/ - "#AARRGGBB" or "#RRGGBB"
+void GpxParser::ParseOsmandColor(std::string const & value) {
+  auto const colorBytes = FromHex(value.substr(1, value.size() - 1));
+  if (colorBytes.size() != 3 && colorBytes.size() != 4)
+  {
+    LOG(LWARNING, ("Invalid color value", value));
+    return;
+  }
+  uint32_t color = 0;
+  if (colorBytes.size() == 3)
+  {
+    color = kml::ToRGBA(colorBytes[0], colorBytes[1], colorBytes[2], (char)255);
+  }
+  else if (colorBytes.size() == 4)
+  {
+    color = kml::ToRGBA(colorBytes[1], colorBytes[2], colorBytes[3], colorBytes[0]);
+  }
```

Comment:
Is it possible to rewrite this block using `switch`?

Reply:
Fixed